File: domain/src/main/java/es/santirivera/surveilfall/domain/usecases/base/UseCasePartialCallback.java
package es.santirivera.surveilfall.domain.usecases.base;

import android.util.Log;

public abstract class UseCasePartialCallback<Ok, Error extends StringErrorOutput> implements UseCaseTaggedCallback<Ok, Error> {

    @Override
    public void onError(String tag, Error error) {
        Log.e("onError", error.getErrorDesc());
    }

    @Override
    public void onGenericError(String tag) {
        Log.e("onGenericError", tag);
    }

    @Override
    public void onNetworkUnavailable(String tag) {
        Log.e("onNetworkUnavailable", tag);
    }

    @Override
    public boolean isReady() {
        return true;
    }
}

File: domain/src/main/java/es/santirivera/surveilfall/domain/usecases/base/UseCasePartialCallback.kt
package es.santirivera.surveilfall.domain.usecases.base

import android.util.Log

abstract class UseCasePartialCallback<Ok, Error : StringErrorOutput> : UseCaseTaggedCallback<Ok, Error> {

    override fun onError(tag: String, error: Error) {
        Log.e("onError", error.errorDesc)
    }

    override fun onGenericError(tag: String) {
        Log.e("onGenericError", tag)
    }

    override fun onNetworkUnavailable(tag: String) {
        Log.e("onNetworkUnavailable", tag)
    }
}
